src/services/catalog.service.ts

```diff
@@ -96,25 +96,39 @@ export class CatalogService {
     return new TemplateItem(resource, this);
   }
 
-  public getCategoriesBySubCategories(tags: any) {
+  public getCategoriesBySubCategories(itemTags: any) {
     let catsBySubCats = {};
     let otherId = 'other';
-
-    _.each(tags, (tag) => {
-      _.each(this.categories, (category) => {
-        let subCat: any = _.find(category.subCategories, (subCategory: any) => {
-              return subCategory.id === tag || _.includes(subCategory.categoryAliases, tag);
-            }
-        );
-        if (subCat) {
-          catsBySubCats[subCat.id] = category.id;
-          return false;
+    let filteredSubCats: any;
+
+    _.each(this.categories, (category) => {
+      if (category.tags) {
+        if (!_.isEmpty(this.getMatchingTags(category.tags, itemTags))) {
+          filteredSubCats = this.filterSubCatsByTags(category.subCategories, itemTags);
+          if (!_.isEmpty(filteredSubCats)) {
+            _.each(filteredSubCats, (subCat) => {
+              catsBySubCats[subCat.id] = category.id;
+            });
+          } else {
+            catsBySubCats[otherId] = category.id;
```

Comment:
Similar case that is broken:

categories:
```
{id: 'mycata', label: 'My Category A', subCategories: [
    {id: 'database', label: 'Database', tags: ['database']}
  ]},
  {id: 'mycatb', label: 'My Category B', subCategories: [
    {id: 'database', label: 'Database', tags: ['database']}
  ]}
```

```
"mycat-a-other": {
    kind: 'ServiceClass',
    "metadata": {
      "name": "mycat-a-other",
      "uid": "11",
    },
    osbTags: [
      "database"
    ]
  }
```

This should show up under both `My Category A -> Database` AND `My Category B -> Database`